projects/forge-angular/src/lib/dialog/dialog.service.ts

```diff
@@ -92,4 +107,41 @@ export class DialogService {
     dialogInstance.open = false;
     ref.destroy();
   }
+
+  public ngOnDestroy(): void {
+    this._unsubscribe$.next(null);
+    this._unsubscribe$.complete();
+  }
+
+  /**
+   * Closes all dialogs. Will close up to 2 nested dialogs.
+   * @param result The result of closing the dialogs. Default is false.
+   */
+  public closeAllDialogs(result = false): void {
+    this._closeAllDialogs(result);
+  }
+
+  private _closeAllDialogs(result: boolean, recursiveExecutionCount = 0): void {
+    if (recursiveExecutionCount > 2) {
```

Comment:
The implementation I referenced used this to prevent a recursive loop.  In our use, we would never have a dialog open another.  In my local implementation of this it is not included, however I recognize there could be instance where a user could have nested dialogs.